src/transforms/filter.jl

```diff
@@ -0,0 +1,51 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Filter(f, table)
+
+The transform that filters the rows based on a given function.
+"""
+struct Filter <: Transform
+    f::Base.Callable
+
+
+    function Filter(new_filter)
+        new(new_filter)
+    end
+
+    function Filter()
+        new(_default_filter)
+    end
+end
+
+isrevertible(::Type{Filter}) = true
+
+#Ideally, this function would also work without a given function, as follows:
+#[item for item in iterable if item]
+function apply(transform::Filter, table)
+    indices = [i for i in range(1,length(table)) if transform.f(table[i])]
```

Comment:
`length` is not a valid function in the Tables.jl API. Your tests are failing on Linux, probably because of that.